uber/api.py

```diff
@@ -281,28 +281,45 @@ def search(self, query, full=False):
 
     def export(self, query, full=False):
         """
-        Searches for attendees by either email or first and last name.
+        Searches for attendees by either email, "first last" name, or
+        "first last &lt;email&gt;" combinations.
 
-        `query` should be a comma or newline separated list of emails and
-        "first last" name combos.
+        `query` should be a comma or newline separated list of email/name
+        queries.
+
+        Example:
+        <pre>Merrium Webster, only.email@example.com, John Doe &lt;jdoe@example.com&gt;</pre>
 
         Results are returned in the format expected by
         <a href="../import/staff">the staff importer</a>.
         """
-        queries = [s.strip() for s in re.split('[\n,]', normalize_newlines(query)) if s.strip()]
+        _re_name_email = re.compile(r'^\s*(.*?)\s*<\s*(.*?@.*?)\s*>\s*$')
+        _re_sep = re.compile(r'[\n,]')
+        queries = [s.strip() for s in _re_sep.split(normalize_newlines(query)) if s.strip()]
 
         names = dict()
         emails = dict()
+        names_and_emails = dict()
         ids = set()
         for q in queries:
             if '@' in q:
-                emails[Attendee.normalize_email(q)] = q
+                match = _re_name_email.match(q)
+                if match:
+                    name = match.group(1)
+                    email = Attendee.normalize_email(match.group(2))
+                    if name:
+                        first, _, last = [s.strip() for s in name.partition(' ')]
```

Comment:
Does this splitting assume that names will only ever have one first name and one last name?

Reply:
Yeah, this is a bad assumption to make generally. I wanted some way to differentiate attendees with the same email address (something that happens surprisingly often). This is a super simple brute force method that could be done better.